android/src/main/java/cl/json/ShareFiles.java

```diff
@@ -150,8 +151,8 @@ private String getRealPathFromURI(Uri contentUri) {
                 String encodedImg = uri.getSchemeSpecificPart().substring(uri.getSchemeSpecificPart().indexOf(";base64,") + 8);
                 try {
                     File dir = new File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_DOWNLOADS );
-                    if (!dir.exists()) {
-                        dir.mkdirs();
```

Comment:
Maybe it is better to keep this?

Reply:
Same as above, this one was actually reinforced by not only being kept but by having it's return value actually used. In fact - ShareFiles and ShareFile should almost inherit from each other, or re-use in some other preferred OO way - the code between them is very duplicate. That's a big change though - much more than mechanical / static-analysis de-lint